lib/server.js

```diff
@@ -334,6 +334,50 @@ async function create (log, error, config, routes, db, oauthdb, translator) {
   await server.register(require('hapi-fxa-oauth'))
 
   server.auth.strategy('oauthToken', 'fxa-oauth', config.oauth)
+
+  server.auth.scheme('fxa-oauth-refreshToken', (server, options) => {
+    return {
+      async authenticate (request, h) {
+        try {
+          const refreshTokenId = request.headers.authorization.split(' ')[1];
+          // TODO: verify that first chunk is Bearer?
+          const refreshToken = await oauthdb.checkRefreshToken(refreshTokenId)
+          if (! refreshToken || ! refreshToken.active) {
+            return h.unauthenticated()
+          }
+
+          const credentials = {
+            uid: refreshToken.sub,
+            refreshTokenId: refreshTokenId,
+            tokenVerified: true
+          }
+
+          const devices = await db.devices(refreshToken.sub)
```

Comment:
I believe there's a helper on the `request` object for fetching the list of devices, to ensure we only do it once per request. Can we use that here?

Reply:
it had an issue, i can try again

Reply:
i glanced at it, it seems like we need to adjust the existing helper t accept the uid argument from the refresh token, seems a bit too much work instead of just calling function directly.